pytorch_lightning/utilities/distributed.py

```diff
@@ -345,3 +346,26 @@ def register_ddp_comm_hook(
 
 def tpu_distributed() -> bool:
     return _TPU_AVAILABLE and xm.xrt_world_size() > 1
+
+
+def init_ddp_connection(
+    cluster_environment: ClusterEnvironment,
+    torch_distributed_backend: str,
+    global_rank: Optional[int] = None,
+    world_size: Optional[int] = None,
+) -> None:
+    global_rank = global_rank if global_rank is not None else cluster_environment.global_rank()
+    world_size = world_size if world_size is not None else cluster_environment.world_size()
+    os.environ["MASTER_ADDR"] = cluster_environment.master_address()
+    os.environ["MASTER_PORT"] = str(cluster_environment.master_port())
+    if torch.distributed.is_available() and not torch.distributed.is_initialized():
```

Comment:
should we raise a misconfiguration error if torch distributed isn't available?

Reply:
I think that's changing the behavior, we are not crashing it rn when we run it on an unsupported platform like Windows.

Reply:
If we decide to chance behaviour, let's do it in another PR.